test/gcp-residency.test.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {strict as assert} from 'assert';
+import {execSync} from 'child_process';
+import * as fs from 'fs';
+
+import {beforeEach, describe, it} from 'mocha';
+
+import * as gcpResidency from '../src/gcp-residency';
+
+function getLinuxBiosVendor() {
+  if (process.platform !== 'linux') return '';
+
+  try {
+    return fs.readFileSync(
+      gcpResidency.GCE_LINUX_BIOS_PATHS.BIOS_VENDOR,
+      'utf8'
+    );
+  } catch {
+    return '';
+  }
+}
+
+function getWindowsBIOSManufacturer() {
+  if (process.platform !== 'win32') return '';
+
+  const query = 'Get-WMIObject -Query "SELECT Manufacturer FROM Win32_BIOS"';
+  return execSync(query, {shell: 'powershell.exe'}).toString();
+}
```

Comment:
[nit]: it could be simpler to mock the `execSync` function and return a deterministic result regardless of platform in order to test this method.